Automattic/jetpack
# [Boost] Add deactivate hook to Cache Module

This PR adds a deactivate hook and code to cleanup when Boost is deactivated.

It does 3 things:
* Removes the WP_CACHE from wp-config.php
* Deletes advanced-cache.php if it's the Boost one.
* Deletes the wp-content/boost-cache directory and sub directories to remove any cached files.

Something to consider - when we add a config file in wp-content/boost-cache/ it should be left there when the plugin is deactivated. If the plugin is uninstalled, then it should be deleted.

Fixes [#35204](https://github.com/Automattic/jetpack/issues/35204)

## Proposed changes:
* Adds new deactivate function to the Page_Cache class.
* Adds a Boost_Cache_Utils class that contains the delete_directories() function. This will be shared with the caching engine, eventually.

### Other information:


## Jetpack product discussion
pc9hqz-2rA-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Patch plugin with this PR.
* Make sure that caching is activated and cache files are in wp-content/boost-cache/cache/
* Deactivate the plugin on the Plugins page.
* Check wp-config.php for WP_CACHE.
* Check if wp-content/advanced-cache.php and wp-content/boost-cache/ have been deleted.
